autotest/gdrivers/vrtprocesseddataset.py

```diff
@@ -0,0 +1,1258 @@
+    <ProcessingSteps>
+        <Step>
+            <Algorithm>LocalScaleOffset</Algorithm>
+            <Argument name="gain_dataset_filename_1">{gain_filename}</Argument>
+            <Argument name="gain_dataset_band_1">1</Argument>
+            <Argument name="gain_dataset_filename_2">{gain_filename}</Argument>
+            <Argument name="gain_dataset_band_2">2</Argument>
+            <Argument name="offset_dataset_filename_1">{offset_filename}</Argument>
+            <Argument name="offset_dataset_band_1">1</Argument>
+            <Argument name="offset_dataset_filename_2">{offset_filename}</Argument>
+            <Argument name="offset_dataset_band_2">2</Argument>
+            <Argument name="min">2</Argument>
+            <Argument name="max">16</Argument>
+        </Step>
+    </ProcessingSteps>
+    </VRTDataset>
+        """
+    )
+    assert struct.unpack("B" * 6, ds.GetRasterBand(1).ReadRaster()) == (
+        2,
+        6,
+        15,
+        255,
+        255,
+        255,
+    )
+    assert struct.unpack("B" * 6, ds.GetRasterBand(2).ReadRaster()) == (
+        2,
+        7,
+        16,
+        255,
+        255,
+        255,
+    )
+
+
+###############################################################################
+# Test nominal case of LocalScaleOffset algorithm where gain and offset have a lower
+# resolution than the input dataset
+
+
+def test_vrtprocesseddataset_dehazing_different_resolution(tmp_vsimem):
+
+    src_filename = str(tmp_vsimem / "src.tif")
+    src_ds = gdal.GetDriverByName("GTiff").Create(src_filename, 6, 2, 1)
+    src_ds.GetRasterBand(1).WriteRaster(0, 0, 6, 2, b"\x01\x01\x02\x02\x03\x03" * 2)
+    src_ds.SetGeoTransform([0, 0.5, 0, 0, 0, 0.5])
+    src_ds.Close()
+
+    gain_filename = str(tmp_vsimem / "gain.tif")
+    gain_ds = gdal.GetDriverByName("GTiff").Create(gain_filename, 3, 1, 1)
+    gain_ds.GetRasterBand(1).WriteRaster(0, 0, 3, 1, b"\x02\x04\x06")
+    gain_ds.SetGeoTransform([0, 1, 0, 0, 0, 1])
+    gain_ds.Close()
+
+    offset_filename = str(tmp_vsimem / "offset.tif")
+    offset_ds = gdal.GetDriverByName("GTiff").Create(offset_filename, 3, 1, 1)
+    offset_ds.GetRasterBand(1).WriteRaster(0, 0, 3, 1, b"\x01\x02\x03")
+    offset_ds.SetGeoTransform([0, 1, 0, 0, 0, 1])
+    offset_ds.Close()
+
+    ds = gdal.Open(
+        f"""<VRTDataset subclass='VRTProcessedDataset'>
+    <Input>
+        <SourceFilename>{src_filename}</SourceFilename>
+    </Input>
+    <ProcessingSteps>
+        <Step>
+            <Algorithm>LocalScaleOffset</Algorithm>
+            <Argument name="gain_dataset_filename_1">{gain_filename}</Argument>
+            <Argument name="gain_dataset_band_1">1</Argument>
+            <Argument name="offset_dataset_filename_1">{offset_filename}</Argument>
+            <Argument name="offset_dataset_band_1">1</Argument>
+        </Step>
+    </ProcessingSteps>
+    </VRTDataset>
+        """
+    )
+    assert struct.unpack("B" * 12, ds.GetRasterBand(1).ReadRaster()) == (
```

Comment:
I think it would improve readability to use numpy for new tests. Is there a packager/user who needs to run the test suite and has sufficiently recent versions of `pytest` etc. but doesn't have `numpy` available?

Reply:
> I think it would improve readability to use numpy for new tests. Is there a packager/user who needs to run the test suite and has sufficiently recent versions of `pytest` etc. but doesn't have `numpy` available?

yes, in that instance, that's probably not necessary to be that conservative. I've added https://github.com/OSGeo/gdal/pull/9598/commits/174c886a340f7a95a08eb8d8dc6c13515d32dd7f that uses numpy functionality